src/device/usbd.c

```diff
@@ -867,10 +868,6 @@ void dcd_event_handler(dcd_event_t const * event, bool in_isr)
     break;
 
     case DCD_EVENT_XFER_COMPLETE:
-      // skip zero-length control status complete event, should DCD notify us.
-      // TODO could cause issue with actual zero length data used by class such as DFU
-      if ( (0 == tu_edpt_number(event->xfer_complete.ep_addr)) && (event->xfer_complete.len == 0) ) break;
-

```

Comment:
Removing this causes multi-packet transfers (i.e. device descriptor readback) to fail on STM32F0. Debugging is in process.

Reply:
I think it is probably the issue with fsdev, this only exposes this. Not entirely sure how this could cause this.